android/src/main/kotlin/id/passage/passage_flutter/PassageFlutter.kt

```diff
@@ -34,12 +36,20 @@ internal class PassageFlutter(private val activity: Activity, appId: String? = n
 
     // region PASSKEY METHODS
 
-    internal fun register(call: MethodCall, result: MethodChannel.Result) {
+    internal fun registerWithPasskey(call: MethodCall, result: MethodChannel.Result) {
         val identifier = call.argument<String>("identifier")
             ?: return invalidArgumentError(result)
         CoroutineScope(Dispatchers.IO).launch {
             try {
-                val authResult = passage.registerWithPasskey(identifier)
+                var options: PasskeyCreationOptions? = null
+                call.argument<Map<String, String>>("options")?.let { map ->
+                    map["authenticatorAttachment"]?.let { authAttachmentString ->
+                        AuthenticatorAttachment.decode(authAttachmentString)?.let { authenticatorAttachment ->
+                            options = PasskeyCreationOptions(authenticatorAttachment)
+                        }
+                    }
+                }
+                val authResult = passage.registerWithPasskey(identifier, options)
```

Comment:
Everything looks good. I just have one question: if 'options' is null, will 'registerWithPasskey' throw an error?

Reply:
Nope, null is acceptable. 👍